aws/bearer/token.go

```diff
@@ -0,0 +1,49 @@
+package bearer
```

Comment:
**Discuss**: Is this where we want this to live? I can understand @isaiahvita point about it not being AWS Credentials (although if you take "credentials" more generically to mean any kind of credential then `aws/credentials/bearer` would be an _ok_ package). This new package hierarchy doesn't make sense either though (`aws/bearer`). Perhaps a new top level `auth` package (`auth/bearer`)?

Reply:
additionally, if we look at this file
https://github.com/aws/aws-sdk-go/blob/main/aws/credentials/credentials.go#L74

the naming of `Value` seems to imply that its definition is what a "credentials" is. if it were to leave open the possibility of other types of credentials we would need a name for the credentials that is composed of "awsaccesskey,awssecretkey,awssessiontoken": so it would be something like `AWSCredentials` and `TokenCredentials`